xbmc/utils/EGLFence.cpp

```diff
@@ -0,0 +1,68 @@
+/*
+ *  Copyright (C) 2017-2018 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "EGLFence.h"
+
+#include "EGLUtils.h"
+
+CEGLFence::CEGLFence(EGLDisplay display) :
+  m_display(display)
+{
+  m_eglCreateSyncKHR = CEGLUtils::GetRequiredProcAddress<PFNEGLCREATESYNCKHRPROC>("eglCreateSyncKHR");
+  m_eglDestroySyncKHR = CEGLUtils::GetRequiredProcAddress<PFNEGLDESTROYSYNCKHRPROC>("eglDestroySyncKHR");
+  m_eglGetSyncAttribKHR = CEGLUtils::GetRequiredProcAddress<PFNEGLGETSYNCATTRIBKHRPROC>("eglGetSyncAttribKHR");
+}
+
+bool CEGLFence::CreateFence()
+{
+  m_fence = m_eglCreateSyncKHR(m_display, EGL_SYNC_FENCE_KHR, nullptr);
+  if (m_fence == EGL_NO_SYNC_KHR)
+  {
+    CEGLUtils::LogError("failed to create egl sync fence");
+    return false;
```

Comment:
What use is this return value?

Reply:
you're right, I don't actually deal with it anywhere. I guess I really only need to log the failure and change bool to void?

Reply:
The question that should be always asked in cases like this is: Is failure expected? I.e., under what circumstances can `eglCreateSyncKHR` fail? Going further: If it fails, can the rest of the code run as normal or will it break somewhere? If failure is expected, there may or may not be any need to log a message, I'll leave that up to you. If it is not expected, why not throw an exception?

I'm a big fan of hard failing by the way. In lots of places in the Kodi code we log a warning and continue doing *something* where we should really just be throwing an exception. Hard-failing allows you to catch bugs sooner because they don't get lost in the log noise.

Reply:
I read the docs and it seems like the only times a fence cannot be created are because the EGLDisplay is wrong or the EGLContext changed. These are pretty large errors so let's hard fail for now. I could never get the create fence call to fail in my limited testing.